build_tools/cmake/iree_llvm.cmake

```diff
@@ -200,8 +200,7 @@ macro(iree_llvm_set_bundled_cmake_options)
   if(IREE_TARGET_BACKEND_METAL_SPIRV)
     message(STATUS "  - metal-spirv")
   endif()
-  if(IREE_TARGET_BACKEND_ROCM)
-    message(STATUS "  - rocm")
+  if(IREE_TARGET_BACKEND_ROCM OR IREE_HAL_DRIVER_AMDGPU)
     list(APPEND LLVM_TARGETS_TO_BUILD AMDGPU)
     set(IREE_CLANG_TARGET clang)
```

Comment:
Does `IREE_HAL_DRIVER_AMDGPU` require `IREE_BUILD_COMPILER`?

Reply:
It only requires clang/lld/llvm-link, so not the full compiler. It's messy today and needs a good cleanup: ideally we don't build anything if the user provides a clang that can compile for the archs they want to target the device library for (`IREE_HAL_AMDGPU_DEVICE_LIBRARY_TARGETS`). Probably needs some cmake compile tests during configuration and only then fallback unless the user already has `IREE_BUILD_COMPILER=ON`. For now I've got `-DIREE_BUILD_COMPILER=ON` + `-DIREE_TARGET_BACKEND_ROCM=ON` in the instructions.

Reply:
(your help would be greatly appreciated on this now that you're in TheRock land and may know how to best integrate with those releases/etc :)